security-malware-firewall.php

```diff
@@ -2450,3 +2452,32 @@ function spbc_private_list_add( $ip ) {
 	}
 	return false;
 }
+
+/**
+ * Check filtered variable to get the api key.
+ *
+ * @return void
+ */
+function spbc_activation__maybe_get_api_key()
+{
+	global $spbc;
+	if (
+		$spbc->api_key ||
+		( ! is_main_site() && $spbc->network_settings['ms__work_mode'] != 2 )
+	) {
+		return;
+	}
+	/**
+	 * Filters a getting API key flag
+	 *
+	 * @param bool Set true if you want to get key automatically after activation the plugin
+	 */
+	$is_get_api_key = apply_filters('spbc_is_get_api_key', false);
+	if ( $is_get_api_key ) {
+		$get_key = spbc_get_key_auto(true);
+		if ( empty($get_key['error']) ) {
+			$spbc->data['key_changed'] = true;
+			$spbc->save('data');
+		}
```

Comment:
This code already contained in spbc_get_key_auto() no need to duplicate it.

Reply:
Done https://github.com/CleanTalk/security-malware-firewall/commit/1393b99fed52e41e73392ada1ebe3f226b39be3e